src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/SpecimenMapper.java

```diff
@@ -1,14 +1,87 @@
 package uk.nhs.digital.nhsconnect.lab.results.translator.mapper;
 
+import lombok.RequiredArgsConstructor;
+import org.hl7.fhir.dstu3.model.Annotation;
+import org.hl7.fhir.dstu3.model.Identifier;
 import org.hl7.fhir.dstu3.model.Specimen;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.FreeTextSegment;
 import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Message;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.Reference;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.SpecimenCharacteristicType;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.SpecimenQuantity;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.segmentgroup.SpecimenDetails;
+import uk.nhs.digital.nhsconnect.lab.results.utils.UUIDGenerator;
 
+import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
+
+import static java.util.stream.Collectors.toList;
 
 @Component
+@RequiredArgsConstructor(onConstructor_ = @Autowired)
 public class SpecimenMapper {
+    private static final String IDENTIFIER_SYSTEM = "http://ehr.acme.org/identifiers/collections";
+    private static final String ACCESSION_IDENTIFIER_SYSTEM = "http://lab.acme.org/specimens/2011";
+    private static final String TYPE_SYSTEM = "http://snomed.info/sct";
+
+    private final UUIDGenerator uuidGenerator;
+    private final DateFormatMapper dateFormatMapper;
+
     public List<Specimen> mapToSpecimens(final Message message) {
-        return List.of(new Specimen());
+        return message.getServiceReportDetails().getSubject().getSpecimens().stream()
+            .map(this::edifactToFhir)
```

Comment:
Can we modify the `testEdifactIsMappedToFhirBundle` test in `EdifactToFhirServiceTest.java` to check that a Specimen is being added to the Bundle?

Reply:
Looking at that test again, it looks to me like it's doing more than it needs to. The method is just pass-through: by which I mean it takes the parameter and passes it straight on to another unit, and then takes the result of that call and passes the result straight into another unit, and then the result of that second call is exactly what is returned. The logic of this method is minimal.

So I see two approaches to verify those pass-throughs: "same as" or "equals".

**1. same as** is the simplest check and verifies that the objects we pass in and return from mocks are the ones we continue through the chain and in the ultimate return, but it doesn't care if that object gets changed in the meanwhile.
**2. equals** is closer to what we have already: it doesn't care if the actual objects being passed around are the same in memory, only that the information they hold is the same - so the method under test could be cloning the input or using some other pattern to construct a _different_ object with the same values and the test would still pass.

I would argue that the responsibility of the `EdifactToFhirService` is just to coordinate the use of the `PathologyRecordMapper` and `BundleMapper` and _to do nothing else_. I would like to change the test with the **1  same as** approach, and also verify that the objects are not being manipulated at all. Then if we start manipulating them, the test will fail, and we can re-evaluate how we approach the test.

Reply:
I'm happy with using **same as** if others are

Reply:
Now changed